components/director/internal/open_resource_discovery/service_test.go

```diff
@@ -0,0 +1,1026 @@
+			vendorSvcFn:  successfulVendorCreate,
+			tombstoneSvcFn: func() *automock.TombstoneService {
+				tombstoneSvc := &automock.TombstoneService{}
+				tombstoneSvc.On("ListByApplicationID", txtest.CtxWithDBMatcher(), appID).Return(nil, nil).Once()
+				ts := fixSanitizedORDDocument().Tombstones[0]
+				ts.OrdID = event1ORDID
+				tombstoneSvc.On("Create", txtest.CtxWithDBMatcher(), appID, *ts).Return("", nil).Once()
+				tombstones := fixTombstones()
+				tombstones[0].OrdID = event1ORDID
+				tombstoneSvc.On("ListByApplicationID", txtest.CtxWithDBMatcher(), appID).Return(tombstones, nil).Once()
+				return tombstoneSvc
+			},
+			clientFn: func() *automock.Client {
+				client := &automock.Client{}
+				doc := fixORDDocument()
+				doc.Tombstones[0].OrdID = event1ORDID
+				client.On("FetchOpenResourceDiscoveryDocuments", txtest.CtxWithDBMatcher(), baseURL).Return(open_resource_discovery.Documents{doc}, nil)
+				return client
+			},
+		},
+		{
+			Name:            "Does not resync resources if vendor resource deletion due to tombstone fails",
+			TransactionerFn: secondTransactionNotCommited,
+			appSvcFn:        successfulAppList,
+			webhookSvcFn:    successfulWebhookList,
+			bundleSvcFn:     successfulBundleCreate,
+			apiSvcFn:        successfulAPICreate,
+			eventSvcFn:      successfulEventCreate,
+			packageSvcFn:    successfulPackageCreate,
+			productSvcFn:    successfulProductCreate,
+			vendorSvcFn: func() *automock.VendorService {
+				vendorSvc := &automock.VendorService{}
+				vendorSvc.On("ListByApplicationID", txtest.CtxWithDBMatcher(), appID).Return(nil, nil).Once()
+				vendorSvc.On("Create", txtest.CtxWithDBMatcher(), appID, *sanitizedDoc.Vendors[0]).Return("", nil).Once()
+				vendorSvc.On("ListByApplicationID", txtest.CtxWithDBMatcher(), appID).Return(fixVendors(), nil).Once()
+				vendorSvc.On("Delete", txtest.CtxWithDBMatcher(), vendorORDID).Return(testErr).Once()
+				return vendorSvc
+			},
+			tombstoneSvcFn: func() *automock.TombstoneService {
+				tombstoneSvc := &automock.TombstoneService{}
+				tombstoneSvc.On("ListByApplicationID", txtest.CtxWithDBMatcher(), appID).Return(nil, nil).Once()
+				ts := fixSanitizedORDDocument().Tombstones[0]
+				ts.OrdID = vendorORDID
+				tombstoneSvc.On("Create", txtest.CtxWithDBMatcher(), appID, *ts).Return("", nil).Once()
+				tombstones := fixTombstones()
+				tombstones[0].OrdID = vendorORDID
+				tombstoneSvc.On("ListByApplicationID", txtest.CtxWithDBMatcher(), appID).Return(tombstones, nil).Once()
+				return tombstoneSvc
+			},
+			clientFn: func() *automock.Client {
+				client := &automock.Client{}
+				doc := fixORDDocument()
+				doc.Tombstones[0].OrdID = vendorORDID
+				client.On("FetchOpenResourceDiscoveryDocuments", txtest.CtxWithDBMatcher(), baseURL).Return(open_resource_discovery.Documents{doc}, nil)
+				return client
+			},
+		},
+		{
+			Name:            "Does not resync resources if product resource deletion due to tombstone fails",
```

Comment:
The naming of the test cases is a bit confusing. If I understand it correctly it should be something like: 
`Does not resync resources if product resource deletion fails due to tombstone`

Reply:
The context is that the product resource should be deleted because there is a tombstone resource for it (that's the `due to tombstone` part of the name). And the test case is that the product deletion fails - everything with the tombstone is ok.

You can think of it as something along the lines of: `Does not resync resources if there's a tombstone for a product resource and the product deletion fails`